interfaces/builtin/maliit_input_method.go

```diff
@@ -0,0 +1,155 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/snapcore/snapd/interfaces"
+)
+
+var maliitInputMethodPermanentSlotAppArmor = []byte(`
+# Description: Allow operating as a maliit server.
+
+# DBus accesses
+#include <abstractions/dbus-session>
+
+/usr/share/glib-2.0/schemas/ r,
+
+# maliit uses peer-to-peer dbus over a unix socket after address negotiation
+# each application has its own one-to-one communication channel with the maliit
+# server, over which all further communication happens.
+unix (bind, listen, accept) type=stream addr="@/tmp/maliit-server/dbus-*",
+`)
+
+var maliitInputMethodConnectedSlotAppArmor = []byte(`
+# Provides the maliit address service which assigns an individual unix socket
+# to each application
+dbus (send, receive)
+    bus=session
+    interface="org.maliit.Server.Address"
+    path=/org/maliit/server/address
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+# provide access to the peer-to-peer dbus socket assigned by the address service
+unix (receive, send) type=stream addr="@/tmp/maliit-server/dbus-*" peer=(label=###PLUG_SECURITY_TAGS###),
+`)
+
+var maliitInputMethodConnectedPlugAppArmor = []byte(`
+# Description: Allow applications to connect to a maliit socket
+# Usage: common
+
+#include <abstractions/dbus-session>
+
+# Allow applications to communicate with the maliit address service
+# which assigns an individual unix socket for all further communication
+# to happen over.
+dbus (send, receive)
+    bus=session
+    interface="org.maliit.Server.Address"
+    path=/org/maliit/server/address
+    peer=(label=###SLOT_SECURITY_TAGS###),
+
+# provide access to the peer-to-peer dbus socket assigned by the address service
+unix (send, receive, connect) type=stream addr=none peer=(label=###SLOT_SECURITY_TAGS###, addr="@/tmp/maliit-server/dbus-*"),
```

Comment:
In thinking about this more, because this is a glob, it allows clients to connect to other client's abstract sockets. Can this be made snap specific? Eg:
```
unix (send, receive, connect) type=stream addr=none peer=(label=###SLOT_SECURITY_TAGS###, addr="@/tmp/maliit-server/dbus-${SNAP_NAME}-*"),
```

If not, what protections are in place when a malicious client tries to use an addressed not assigned to it? Please update the preceding comment to discuss this.